sensing/autoware_pointcloud_preprocessor/src/downsample_filter/pickup_based_voxel_grid_downsample_filter_node.cpp

```diff
@@ -116,19 +112,22 @@ void PickupBasedVoxelGridDownsampleFilterComponent::filter(
       static_cast<int>((x + large_num_offset) * inverse_voxel_size_x),
       static_cast<int>((y + large_num_offset) * inverse_voxel_size_y),
       static_cast<int>((z + large_num_offset) * inverse_voxel_size_z)};
-
-    voxel_map.emplace(key, global_offset);
+    index_map.emplace(key, point_index);
   }
+}
 
-  // Populate the output point cloud
+void copy_filtered_point(
```

Comment:
Since this copies multiple points, the name should be plural:
```suggestion
void copy_filtered_points(
```